### DISCUSSION TITLE
Nextjs 14 and Server Components support plans?!

### DISCUSSION THREAD

**Discussion by SharadKumar** [tinacms/tinacms#4404] [General]
Hi Team, I am wondering if there is an upcoming plan for supporting contextual editing and blocks with Server Components, App router / Nextjs 13/14?

I couldnt find explicit statement of Nextjs supported versions, although all examples are only Nextjs v12.

TinaCMs is very promising and it would be of great value if this could work with current versions.

**Comment by arunavo4** [comment]
@SharadKumar You can use server components and also use the app router with the current setup. You can use this as the starting point.
https://github.com/twittkopf/tinacms-visual-editing-nextjs-app-dir
